trunk/SUAVE/Components/Energy/Networks/Battery_Propeller.py

```diff
@@ -90,151 +98,263 @@ def evaluate_thrust(self,state):
-            P[eta>1.0] = P[eta>1.0]*eta[eta>1.0]
-            F[eta[:,0]>1.0,:] = F[eta[:,0]>1.0,:]*eta[eta[:,0]>1.0,:]
-
-            # Run the motor for current
-            motor.current(conditions)
-            
-            # Conditions specific to this instantiation of motor and propellers
-            R                   = prop.tip_radius
-            rpm                 = motor.outputs.omega / Units.rpm
-            F_mag               = np.atleast_2d(np.linalg.norm(F, axis=1)).T
-            total_thrust        = total_thrust + F * factor
-            total_power         = total_power  + P * factor
-            total_motor_current = total_motor_current + factor*motor.outputs.current
+            dV_TH_dt =  np.dot(D,V_Th_cell)
+            I_cell   = V_Th_cell/(R_Th_cell * battery.R_growth_factor)  + C_Th_cell*dV_TH_dt
+            R_0_cell = R_0_cell * battery.R_growth_factor
+             
+            # Voltage under load
+            volts =  n_series*(V_oc_cell - V_Th_cell - (I_cell  * R_0_cell)) 
 
-            # Pack specific outputs
-            conditions.propulsion.propeller_motor_torque[:,ii] = motor.outputs.torque[:,0]
-            conditions.propulsion.propeller_torque[:,ii]       = Q[:,0]
-            conditions.propulsion.propeller_rpm[:,ii]          = rpm[:,0]
-            conditions.propulsion.propeller_tip_mach[:,ii]     = (R*rpm[:,0]*Units.rpm)/a[:,0]
-            conditions.propulsion.disc_loading[:,ii]           = (F_mag[:,0])/(np.pi*(R**2)) # N/m^2                  
-            conditions.propulsion.power_loading[:,ii]          = (F_mag[:,0])/(P[:,0])      # N/W      
-            conditions.propulsion.propeller_efficiency         = etap[:,0]      
-            
-            if n_evals==1:
-                # Append outputs to each identical propeller
-                for i,p in enumerate(props):
-                    conditions.noise.sources.propellers[p.tag]      = outputs
+        elif type(battery) == SUAVE.Components.Energy.Storages.Batteries.Constant_Mass.Lithium_Ion_LiNiMnCoO2_18650: 
+            SOC        = state.unknowns.battery_state_of_charge 
+            T_cell     = state.unknowns.battery_cell_temperature
+            I_cell     = state.unknowns.battery_current/n_parallel 
+            
+            # Link Temperature 
+            battery.cell_temperature         = T_cell  
+            battery.initial_thevenin_voltage = V_th0   
+            
+            # Compute State Variables
+            V_ul_cell = compute_NMC_cell_state_variables(battery_data,SOC,T_cell,I_cell) 
+            
+            # Voltage under load
+            volts     = n_series*V_ul_cell      
+            
+        # --------------------------------------------------------------------------------
+        # Run Motor, Avionics and Systems (Discharge Model)
+        # --------------------------------------------------------------------------------    
+        if discharge_flag:    
```

Comment:
This is going to be the biggest headache for me on the forum

Reply:
I tried to make this a seamless as possible. Within the networks that use batteries, we need a flag to reverse the current and not run the propeller, motors etc. I changed the flag to "battery_discharge_flag" for clarity, but I do not think - now given I made updates to the segments - a user who reads the code should have any issues.